app/models/piece.rb

```diff
@@ -114,5 +118,31 @@ def obstructed?(x, y)
     end
   end
   
+  def in_check?(king)
+    opposite_pieces = pieces.where(color: !king.color)
+    opposite_pieces.each do |piece|
+      if piece.valid_move?(king.position_x, king.position_y)
+        return true
+      else
+        return false
+      end
+    end
+  end
+  
+  def not_moved_to_different_space?(x,y)
+    position_x == x && y_position == y
+  end
+  
+  def is_my_turn?
+    #write logic to check for whether it's the piece owner's turn here!
+    #maybe something like... @game.turn == current_player.color???? with the turn defined by color elsewhere?
+  end
+  
+  def valid_move?(x,y)
+    #i'm not sure I'm implementing in check correctly here. second set of eyes would be appreciated!
+    obstructed?(x,y) && off_the_board?(x,y) && in_check?(king) && not_moved_to_different_space?(x,y) && is_my_turn?
```

Comment:
in_check? seems to make sense. If the intention is to pass in a king object, which seems to be the idea.